src/patchelf.cc

```diff
@@ -1745,21 +1754,81 @@ void ElfFile<ElfFileParamNames>::addNeeded(const std::set<std::string> & libs)
     changed = true;
 }
 
+// https://stackoverflow.com/a/478960/143733
+std::string exec(const char* cmd) {
+    std::array<char, 128> buffer;
+    std::string result;
+    std::unique_ptr<FILE, decltype(&pclose)> pipe(popen(cmd, "r"), pclose);
+    if (!pipe) {
+        throw std::runtime_error("popen() failed!");
+    }
+    while (fgets(buffer.data(), buffer.size(), pipe.get()) != nullptr) {
+        result += buffer.data();
+    }
+    return result;
+}
+
 template<ElfFileParams>
-void ElfFile<ElfFileParamNames>::printNeededLibs() // const
+void ElfFile<ElfFileParamNames>::shrinkWrap(std::map<std::string, std::string> & neededLibsToReplace, std::set<std::string> & neededLibsToAdd)
+{
+    const std::string interpreter = getInterpreter();
+    const std::vector<std::string> needed = getNeededLibs();
+    std::stringstream ss;
+    ss  << interpreter << " --list " << this->fileName;
+    const std::string cmd = ss.str();
+    const std::string lddOut = exec(cmd.c_str());
+
+    std::istringstream iss(lddOut);
+    std::string line;
+    while (std::getline(iss, line)) {
+        std::regex r("\\s*([^ ]+) => ([^ ]+)");
+        std::smatch matches;
```

Comment:
Not sure about `matches`, but it would make sense to to no recompile the regex on every line:

```suggestion
    std::string line;
    std::regex r("\\s*([^ ]+) => ([^ ]+)");
    std::smatch matches;
    while (std::getline(iss, line)) {

```